joomla/joomla-cms
# [6.0] Move CMS Filesystem package to compat plugin

### Summary of Changes
This PR moves the CMS Filesystem classes to the compat plugin, allowing legacy third party code to still run when the plugin is active, but making the first step to finally remove it from the CMS.

I've added the `Composer Dependency Changed` label because the PR requires `composer i` to be run.

### Testing Instructions
Copy the following line in a view of your choice to generate a call to the legacy classes:
```
var_dump(\Joomla\CMS\Filesystem\Folder::listFolderTree(JPATH_BASE, '.'));
```
Make sure the compat plugin is enabled AND you have to run `composer i` to generate an updated class loader file
Visit the view you edited in the browser to see an output of the call to the legacy class. Disable the plugin and see an error thrown instead.


### Link to documentations
Please select:

- [X] Pull Request link for manual.joomla.org: joomla/Manual/pull/376
